protocol/xmrmaker/swap_state.go

```diff
@@ -424,9 +425,28 @@ func (s *swapState) reclaimMonero(skA *mcrypto.PrivateSpendKey) (mcrypto.Address
 		return "", err
 	}
 
-	s.XMRClient().Lock()
-	defer s.XMRClient().Unlock()
-	return monero.CreateWallet("xmrmaker-swap-wallet", s.Env(), s.XMRClient(), kpAB, s.moneroStartHeight)
+	conf := s.XMRClient().CreateABWalletConf("xmrmaker-swap-wallet-refund")
+	abWalletCli, err := monero.CreateSpendWalletFromKeys(conf, kpAB, s.moneroStartHeight)
+	if err != nil {
+		return "", err
+	}
+	defer abWalletCli.CloseAndRemoveWallet()
+
+	transfers, err := abWalletCli.SweepAll(
+		s.ctx,
+		s.XMRClient().PrimaryAddress(),
+		0,
+		monero.SweepToSelfConfirmations,
+	)
+	if err != nil {
+		return "", fmt.Errorf("failed to sweep refund back to primary wallet: %w", err)
+	}
+	for _, transfer := range transfers {
+		// It is unlikely that anyone will ever see more than one of these messages
+		log.Infof("Refunded %s XMR from swap to primary wallet (%s XMR lost to fees)",
+			coins.FmtPiconeroAmtAsXMR(transfer.Amount), coins.FmtPiconeroAmtAsXMR(transfer.Fee))
+	}
+	return kpAB.Address(s.Env()), nil
```

Comment:
this seems to be basically the same code as `func (inst *Instance) recoverRefund`, can you update these two to use the same function

Reply:
I wanted to do that originally, and it is part of the reason I realized that `SweepAll` should be handling the receipt (which it now does). The issue was that these functions are for objects of different types. One is an `Instance` member function, while the other is a `swapState` function. I made a procedural (non-object-oriented) function named `sweepRefund` and put it in `xmrmaker/utils.go` so the code could be shared. It is a bit of a monster, in that it requires 7 parameters, but it was good to consolidate the code.